source/extensions/filters/http/lua/lua_filter.h

```diff
@@ -329,8 +417,13 @@ class Filter : public Http::StreamFilter, Logger::Loggable<Logger::Id::lua> {
   // Http::StreamDecoderFilter
   Http::FilterHeadersStatus decodeHeaders(Http::RequestHeaderMap& headers,
                                           bool end_stream) override {
+    const auto config_per_route = getConfigPerRoute(decoder_callbacks_.callbacks_);
```

Comment:
I remember having this discussion with @dio a long time ago, but I'm not convinced this works correctly. You are potentially going to be modifying TLS from inside a worker thread vs. the main thread which I don't think works correctly. I'm also a little unsure of resource cleanup in the face of RDS changes. Can you describe how this all works if you have analyzed it?

No matter what please add an integration test for this behavior. I'm pretty positive you will start hitting asserts.

Reply:
Yeah. Sorry for missing to check on this one. I think I had it working back then (with some integration tests), let me check again.